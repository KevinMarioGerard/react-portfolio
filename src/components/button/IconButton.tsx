import * as React from 'react';

type IconButtonProps = {
    /**
     * Icon to render
     */
    icon: string;

    /**
     * Redirection url on button click
     */
    redirectionUrl: string;
};

const IconButton: React.FC<IconButtonProps> = ({ icon, redirectionUrl, ...rest }) => (
    <div
        className="icon icon__container"
        onClick={React.useCallback(() => {
            window.open(redirectionUrl);
        }, [redirectionUrl])}
        {...rest}>
        <i className={`fa ${icon}`} />
    </div>
);

export default IconButton;
