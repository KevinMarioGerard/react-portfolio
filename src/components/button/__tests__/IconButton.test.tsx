import { render, fireEvent } from '@testing-library/react';
import IconButton from '../IconButton';
import * as React from 'react';

describe('Icon Button Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<IconButton icon="" redirectionUrl="" />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<IconButton icon="fa-github" redirectionUrl="" />);

        expect(container).toMatchSnapshot();
    });

    it('should redirect on click', () => {
        window.open = jest.fn();

        const { getByTestId } = render(
            <IconButton data-testid="icon-button" icon="fa-github" redirectionUrl="test-url.com" />
        );

        fireEvent.click(getByTestId('icon-button'));

        expect(window.open).toHaveBeenCalledWith('test-url.com');
    });
});
