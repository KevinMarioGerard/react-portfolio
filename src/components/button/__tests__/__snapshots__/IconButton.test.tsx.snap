// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button tests should match snapshot 1`] = `
<div>
  <div
    class="icon icon__container"
  >
    <i
      class="fa fa-github"
    />
  </div>
</div>
`;
